python/lib/client/dmod/client/dmod_client.py

```diff
@@ -225,7 +229,17 @@ def data_service_client(self) -> DataServiceClient:
             if self.client_config.data_service is not None and self.client_config.data_service.active:
                 t_client_type = determine_transport_client_type(self.client_config.data_service.endpoint_protocol)
                 t_client = t_client_type(**self.client_config.data_service.dict())
-                self._data_service_client = DataServiceClient(t_client, self._auth_client)
+                # use http client where applicable when bypassing request service
+                if self._bypass_request_service:
```

Comment:
This is probably going to need some modifications.  As is, we won't get the extra HTTP client without the user manually specifying an extra CLI arg, and configuring the data service details will have the side effect of bypassing the request service for _all_ data service communication.  I expected that we wouldn't (necessarily) want to bypass the request server except for the actual data transfer.

This `_bypass_request_service` attribute (and related CLI option) is a legacy thing that just never got taken out.   It's `False` unless the user specifically does something extra at runtime.  

What replaced that legacy behavior was the CLI client moving to using a configuration class, and instead bypassing the request service (for data service actions) based on whether a `self.client_config.data_service` value was available.  So if the _ClientConfig_ has a `data_service` value, use that _ConnectionConfig_ when lazily creating the `data_service_client` property.  Otherwise, create `data_service_client` using the "regular" transport client, which was created from `ClientConfig.request_service`.

Depending on how we want to go about things, we'll at least need to change where and how the conditional is used, and possibly more than that if we want to continue to support full bypassing of the request  service (though maybe we don't).

Reply:
> This is probably going to need some modifications. As is, we won't get the extra HTTP client without the user manually specifying an extra CLI arg, and configuring the data service details will have the side effect of bypassing the request service for _all_ data service communication. I expected that we wouldn't (necessarily) want to bypass the request server except for the actual data transfer.

I should have left a note in the code here why I chose to go this way. My thinking was this was the most sensible default for the moment given where we are in the development lifecycle. Namely, in our deployment architecture with the `request` service acting as the main entry point to the deployment, at this point in time, we don't have a way to reach the `http` data service endpoints through the `request` service or via proxy. So, the only way to reach the `http` data service endpoints are by directly reaching out to the data service.

Once this is no longer the case, we will need to modify this section of the client code to remove this condition. Thoughts on that?

#338 is also related.

Reply:
The part that I am having trouble with is the de facto bypassing for any data service communication.   A necessary (but not sufficient) condition for HTTP uploads produces side effects for more than just uploads.  To work around them, we'd have to go back and edit the config (or manually toggle between configs).  

I don't think we want this, and I think it should be fairly easy to avoid.  Any reason not to also make the outer condition dependent on `self._bypass_request_service`?  Or, perhaps more generally, why not (at least for now) change the request service bypassing functionality to once again require an explicit CLI arg, and also make use of HTTP uploads dependent on having bypassed the request service?

Reply:
Just addressed this